src/autocomplete.js

```diff
@@ -0,0 +1,193 @@
+const makePrefixes = regex => {
+  if (regex.length === 0) {
+    throw new Error('empty regex')
+  }
+  else if (regex.length === 1) {
+    return regex
+  }
+  const prefix = regex.charAt(0) === ' ' ? '\\s+' : regex.charAt(0)
+  return `${prefix}(?:${makePrefixes(regex.slice(1))})?`
+}
+
+const regexPrefixes = regex => new RegExp(makePrefixes(regex), 'iu')
+
+const getSuggestions = (exp, suggestions) => {
+  const digitsMatch = exp.match(/\d+/u)
+  const digits = digitsMatch ? digitsMatch[0] : ''
+  const rePrefixes = suggestions.map(regexPrefixes)
+  return suggestions.filter(
```

Comment:
Consider writing:

```javascript
return suggestions
  .filter(...)
  .map(...)
```

It is easier to read and refactor.

Reply:
I disagree, this is easier to read with functional approach but not imperative. Usually, I use both evenmore when I can avoid duplicated code or recursive ones

Reply:
You misunderstood, instead of:

```javascript
return suggestions.filter(
   ...
).map(
  ...
)
```

I propose:

```javascript
return suggestions
  .filter(...)
  .map(...)
```

Reply:
Or even:


```javascript
return suggestions
  .filter(
    ...
  )
  .map(
    ...
  )
```